http-server/auth/auth.go

```diff
@@ -0,0 +1,159 @@
+package authorization
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	http_server "github.com/ZhdanovichVlad/go_final_project/http-server"
+	"github.com/ZhdanovichVlad/go_final_project/storage/users"
+	"github.com/dgrijalva/jwt-go"
+	"net/http"
+	"os"
+	"time"
+)
+
+type Claims struct {
+	Authorized bool `json:"authorized"`
+	Sum        int  `json:"sum"`
+	jwt.StandardClaims
+}
+
+// Authorization function for user authorization.
+// Authorization функция для авторизации пользователя.
+func Authorization(w http.ResponseWriter, req *http.Request) {
+	var user users.User
+	var buf bytes.Buffer
+	_, err := buf.ReadFrom(req.Body)
+	if err != nil {
+		msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"Error when reading from req.Body"}, true)
+		w.WriteHeader(errInt)
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write(msg)
```

Comment:
В идеале обрабатывать ошибки записи ответа
```suggestion
		_, err := w.Write(msg)
		if err != nil {
			log.Printf("error while writing response: %v", err)
		}
```